source/lib.civet

```diff
@@ -1491,21 +1491,26 @@ function hoistRefDecs(statements: StatementTuple[]): void {
     })
 }
 
-function insertHoistDec(block, node, dec): void {
+function insertHoistDec(block: BlockStatement, node: ASTNode, dec): void
   // NOTE: This is more accurately 'statements'
-  const { expressions } = block
-  const index = expressions.findIndex((exp) => exp === node ||
-    (Array.isArray(exp) && exp[1] === node))
-  if (index < 0) throw new Error("Couldn't find expression in block for hoistable declaration.")
-  if (expressions[index] === node) {
-    expressions.splice(index, 0, ["", dec, ";"])
-  } else { // [ws, exp, delim] statement
-    const indent = expressions[index][0]
-    expressions.splice(index, 0, [indent, dec, ";"])
-  }
+  { expressions } := block
+  index := expressions.findIndex (exp) =>
+    exp is node or (Array.isArray(exp) and exp[1] is node)
+
+  if index < 0
+    throw new Error("Couldn't find expression in block for hoistable declaration.")
+
+  let statement
+  if index > 0
```

Comment:
I'm not sure why this condition changed from `if expressions[index] is node`. It seems like the code inside the `if` only works when `expressions[index]` is already an `[indent, exp, delimiter]` array, which I'm guessing `node` is not supposed to be? Whereas the `else` case is when there is no indentation to find. There's no use of `index-1` so `index > 0` doesn't seem relevant.